scripts/m18_submission_checker/aida_m18_submission_checker.py

```diff
@@ -0,0 +1,383 @@
+'''
+This script is designed to parse an archive [.zip, .tgz, .tar.gz] and determine if it is properly structured, and will
+work with Next Century's validator.
+
+To run:
+python3 aida_m18_submission_checker.py <path/to/archive>
+
+Output will be printed to stdout and/or stderr
+'''
+import sys
+import os
+import tarfile, zipfile
+import re
+import logging
+
+
+'''
+Globals
+'''
+
+#variable to store if we use tar or zip method to access the file
+#-1 = error. cannot continue
+#0 = .tar.gz, or .tgz
+#1 = .zip
+FILETYPE = -1
+
+#task types to validate for
+TASK_TYPES = ["1a", "1b", "2", "3"]
+
+#the base name of the archive file
+ARCHIVE_FILE_NAME = ''
+
+#list of names of items in the archive - to be populated after determining what filetype the archive is
+ARCHIVE_NAMES = []
+
+def do_TA1a_check():
+    """
+    Check the archive
+    TASK 1a Directory structure:
+    <TA1aperformer>_<run>
+        NIST
+            <document_id>.ttl                  (1 to X)
+		INTER-TA
+		    <document_id>.ttl                  (0 to X)
+
+    Ignores non ttl files.
+    Prints number of ttl files found, and number of ttl files that are in need to be checked.
+
+    :return: True if it appears submit ready, False otherwise
+    :rtype: bool
+    """
+
+    ttls_found_count = 0
+    ttls_error_warning_count = 0
+    ttls_valid_count = 0
+
+    for name in ARCHIVE_NAMES:
+
+        #split into levels
+        path_items = re.split('/', name)
+
+#TODO CHECK TA1A AND TA1B FOR IF PATH_ITEMS[-1] IS EVER NOT THE LOCATION OF THE .TTL FILE NAME
+
+        if path_items[-1].endswith('.ttl'):
+
+            ttls_found_count += 1
+
+            #validate that the ttl file is in an acceptable directory
+            #rule: ttl needs to be in 2nd level -- aka 3rd item in list['name', 'NIST', 'valid.ttl']
+            #rule: ttl needs to be in directory named NIST, or INTER-TA
+            try:
+                if ((path_items[-2] == 'NIST' or path_items[-2] == 'INTER-TA') and len(path_items) == 3):
+                    ttls_valid_count = ttls_valid_count + 1
+                else:
+                    ttls_error_warning_count += 1
+            except:
+                ttls_error_warning_count += 1
+
+    logging.info("\033[1mTA_1a\033[0m Archive Validator report:")
+    logging.info("TTL files found: " + str(ttls_found_count))
+    logging.info("TTL files in valid locations: " + str(ttls_valid_count))
+    if ttls_error_warning_count > 0:
+        logging.error("TTL files found in an invalid location: " + str(ttls_error_warning_count))
+        logging.error(".ttl files must be located in either the <run_ID>/NIST/ or <run_id>/INTER-TA directory")
+
+    if ttls_found_count > 0 and ttls_error_warning_count == 0:
```

Comment:
I think this was an issue in my previous comment, but I am not sure you want to check that the warning count must be zero here. Just because .ttl files are not in a valid location does not mean that their submission won't be validated. It just means those files will not be included in the validation. @craigwarsaw we should probably talk about this more.

Reply:
See my comment below.  I distinguish between a SUCCESS with all files in the right location and a PARTIAL SUCCESS where it can be validated but there are some files in the wrong location.

Reply:
implemented the distinction, double check proper usage on next commit